src/Leaflet.js

```diff
@@ -1,29 +1,170 @@
+import {Zoom, zoom} from './control/Control.Zoom';
+import {Scale, scale} from './control/Control.Scale';
+import {Attribution, attribution} from './control/Control.Attribution.js';
 
-	window.L = L;
-}
+Control.Layers = Layers;
+Control.Zoom = Zoom;
+Control.Scale = Scale;
+Control.Attribution = Attribution;
 
-// define Leaflet for Node module pattern loaders, including Browserify
-if (typeof module === 'object' && typeof module.exports === 'object') {
-	module.exports = L;
+control.layers = layers;
+control.zoom = zoom;
+control.scale = scale;
+control.attribution = attribution;
 
-// define Leaflet as an AMD module
-} else if (typeof define === 'function' && define.amd) {
-	define(L);
-}
+export {Control, control};
+
+// core
+
+import * as Browser from './core/Browser';
+export {Browser};
+
+export {Class} from './core/Class';
+
+import {Evented} from './core/Events';
+export {Evented};
+// export var Mixin = {Events: Evented.prototype};
+
+export {Handler} from './core/Handler';
+
+import * as Util from './core/Util';
+export {Util};
+export {extend, bind, stamp, setOptions} from './core/Util';
+
+// dom
+
+export {PosAnimation} from './dom/PosAnimation';
+
+import * as DomEvent from './dom/DomEvent';
+export {DomEvent};
+
+import * as DomUtil from './dom/DomUtil';
+export {DomUtil};
+
+export {Draggable} from './dom/Draggable';
+
+// geometry
+
+export {Point, toPoint as point} from './geometry/Point';
+export {Bounds, toBounds as bounds} from './geometry/Bounds';
+export {Transformation} from './geometry/Transformation';
+
+// geo
+
+export {LatLng, toLatLng as latLng} from './geo/LatLng';
+export {LatLngBounds, toLatLngBounds as latLngBounds} from './geo/LatLngBounds';
+
+// geo/projection
+
+import * as Projection from './geo/projection/Projection';
+export {Projection};
+
+// geo/crs
+
+import {CRS} from './geo/crs/CRS';
+import {Earth} from './geo/crs/CRS.Earth';
+import {EPSG3395} from './geo/crs/CRS.EPSG3395';
+import {EPSG3857, EPSG900913} from './geo/crs/CRS.EPSG3857';
+import {EPSG4326} from './geo/crs/CRS.EPSG4326';
+import {Simple} from './geo/crs/CRS.Simple';
+
+CRS.Earth = Earth;
+CRS.EPSG3395 = EPSG3395;
+CRS.EPSG3857 = EPSG3857;
+CRS.EPSG900913 = EPSG900913;
+CRS.EPSG4326 = EPSG4326;
+CRS.Simple = Simple;
+
+export {CRS};
+
+// layer
+
+export {Layer} from './layer/Layer';
+export {LayerGroup, layerGroup} from './layer/LayerGroup';
+export {FeatureGroup, featureGroup} from './layer/FeatureGroup';
+import {GeoJSON, geoJSON, geoJson, geometryToLayer, coordsToLatLng, coordsToLatLngs, latLngToCoords, latLngsToCoords, getFeature, asFeature} from './layer/GeoJSON';
+GeoJSON.geometryToLayer = geometryToLayer;
+GeoJSON.coordsToLatLng = coordsToLatLng;
+GeoJSON.coordsToLatLngs = coordsToLatLngs;
+GeoJSON.latLngToCoords = latLngToCoords;
+GeoJSON.latLngsToCoords = latLngsToCoords;
+GeoJSON.getFeature = getFeature;
+GeoJSON.asFeature = asFeature;
+export {GeoJSON, geoJSON, geoJson};

```

Comment:
Maybe add the static functions to the GeoJSON class in `./layer/GeoJSON.js`? Or prepare and export a `GeoJSON_with_static_functions` object in `./layer/GeoJSON.js`?